### ISSUE TITLE
Braking while taxiing turn autobrake off

### ISSUE THREAD

**Issue by lrargerich** [flybywiresim/a32nx#255]
If you break for traffic while taxiing autobrakes are turned off.

**Comment by wpine215** [comment]
Autobrake low/medium, or autobrake max is turned off by manual braking?

**Comment by PCGamesF1Tech** [comment]
Autobrake should only activate above 72kt (wheel spin) AND spoilers armed.
Autobrake MAX will be selected for T/O (in case of RTO). 
After landing autobrake deactivates once you press the brake pedals (if selected at all during approach preparation).

**Comment by lrargerich** [comment]
While taxiing configured for takeoff (Autobrake Max,Flaps 2, Spoilers On)
If I break on the taxiway autobrake turns off.

**Comment by flightsimsimon** [comment]
I concur - autobreak set for RTO is part of the checklist at the gate, whenever you break it disengages the RTO setting, even at 5knts

**Comment by PCGamesF1Tech** [comment]
The first officer armes autobrake while taxiing.
It's also part of the TO memo/checklist on ECAM. 
It stays armed until after liftoff and gear retraction.

**Comment by flightsimsimon** [comment]
Agreed but auto break disengages soon as you break

**Comment by PCGamesF1Tech** [comment]
Still no, it doesn't in the real aircraft, maybe it does in the sim. 
We set MAX once and taxi further for takeoff without even touching it anymore

**Comment by Curtis-VL** [comment]
TL;DR - Autobrake shouldn't disable when in 'TAKEOFF' phase of the flight.
(Just a note for myself or whoever works on this!)

**Comment by ErikNL80** [comment]
Is the cause already known and a possible solution?

**Comment by geoffda** [comment]
I'm going to investigate this. Trying to figure out how to get it assigned to me.

**Comment by geoffda** [comment]
Problem is in the update code for FBW_Airbus_HANDLING_Push_Autobrakes_Template in A32NX_Interior_Misc.xml

                        <UPDATE_CODE>
                        (A:AUTO BRAKE SWITCH CB, Enum) s1 0 == if{
                            (L:XMLVAR_Autobrakes_Level) 0 == if{
                                #AUTOBRAKE_LEVEL# (&gt;L:XMLVAR_Autobrakes_Level)
                            }
                        } els{
                            l1 1 == if{
                                0 (&gt;L:XMLVAR_Autobrakes_Level)
                            } els{
                                l1 1 - (&gt;L:XMLVAR_Autobrakes_Level)
                                (A:SIM ON GROUND, Bool) (A:GROUND VELOCITY, Knots) 10 &lt; and if{
                                    0 (&gt;K:SET_AUTOBRAKE_CONTROL)
                                }
                            }
                        }
                        </UPDATE_CODE>

The issue is that AUTO BRAKE SWITCH CB isn't really the right simvar for the A320 switches. The behavior of the simvar is that if any level of autobrakes are engaged, it returns 0, which would be RTO if this were the Boeing switch. When autobrakes are off, 1 is returned. When the brakes are pressed during taxi (with the  MAX AUTO BRK button pushed), the state of AUTO BRAKE SWITCH CB changes from 0 (RTO) to 1 (OFF). This results in the else path being taken. Since A:AUTO BRAKE SWITCH CB == 1, that is on the stack, so l1 == 1 and L:XMLVAR_Autobrakes_Level gets set to 0, which turns off autobraking, since L:XMLVAR_Autobrakes_Level is what is actually bound to the autobrake buttons.

So the real question is under what circumstances we need to update the autobrakes setting. I don't see any case where L:XMLVAR_Autobrakes_Level is getting updated externally, so I think the only thing we might want to do is pay attention when autobrakes are actually engaged and then automatically turn them off once the speed falls below a certain level. I have a local fix working right now which turns off the switch after they have been engaged and the speed falls below 10 KIAS, but that may not be correct. I'll see if I can get ahold of the manuals and find out what the real behavior should be and implement that.

**Comment by Benjozork** [comment]
@geoffda we are considering making our own autobrake system - I think @lukecologne wanted to

**Comment by geoffda** [comment]
Also, we probably need to turn off autobrakes once we are in the air as is the current behavior.

**Comment by geoffda** [comment]
@Benjozork, thanks for the heads up. I'd still like to get this fixed before then. It isn't much work and it is an annoying bug (at least to me and some others) :).

**Comment by geoffda** [comment]
The autobrakes should disarm 10 seconds after the landing gear comes up, when the ground spoilers are retracted, or when sufficient braking deflection is applied to one pedal. How much is "sufficient"?

**Comment by geoffda** [comment]
I have a local fix now with much more plausible autobrake behavior. Currently, autobrakes are disarmed immediately upon landing gear retraction, or once autobrakes have disengaged after engaging. Assuming the autobrake system gets rewritten to real life specifications, then the auto brake switches should disarm at the right time.

The only thing I'm missing is the 10 second delay between landing gear retraction and disarming. Now that I've got the basic behavior working, I think I can implement the delay by just sticking a counter in the update code, since we refresh once per second. It should be accurate to +- a second, which I think is good enough.

The only other questions are around edge case scenarios, like a balked landing where the autobrakes are armed, but never engage, yet the plane touches down. I would think that they should remain armed until the landing gear is retracted. I probably need to add a check for that as well.

Anyway, I'm expecting to have a pull request ready by EOD tomorrow.

**Comment by geoffda** [comment]
Well, it looks like refresh rate is more on the order of 3 seconds, so I put a 30 count in which is approximately 10 seconds on my machine. Presumably it will vary a bit, but the behavior should be much closer to reality. Also, I put a check in to prevent setting autobrakes to max in mid-air, which is also according to real behavior.
